x/distribution/keeper/allocation.go

```diff
@@ -0,0 +1,51 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// Allocate fees handles distribution of the collected fees
+func (k Keeper) AllocateFees(ctx sdk.Context) {
+	ctx.Logger().With("module", "x/distribution").Error(fmt.Sprintf("allocation height: %v", ctx.BlockHeight()))
+
+	// get the proposer of this block
+	proposerConsAddr := k.GetProposerConsAddr(ctx)
+	proposerValidator := k.stakeKeeper.ValidatorByConsAddr(ctx, proposerConsAddr)
+	proposerDist := k.GetValidatorDistInfo(ctx, proposerValidator.GetOperator())
+
+	// get the fees which have been getting collected through all the
+	// transactions in the block
+	feesCollected := k.feeCollectionKeeper.GetCollectedFees(ctx)
+	feesCollectedDec := types.NewDecCoins(feesCollected)
+
+	// allocated rewards to proposer
+	percentVotes := k.GetPercentPrecommitVotes(ctx)
+
+	proposerMultiplier := sdk.NewDecWithPrec(1, 2).Add(sdk.NewDecWithPrec(4, 2).Mul(percentVotes))
```

Comment:
Can you add a comment for whats happening right here. Its not clear to me where the number `4` is coming from.

Reply:
as per previous discussions not sure we care to put this in the params store, if we don't expect to change often at-all - I wonder how that change would affect the performance though? This code does execute every block

Reply:
In my opinion, this seems like one of those things that governance should absolutely be able to control.

Reply:
I don't think the performance costs are substantial at all, especially once we have inter-block caching.

Reply:
cool - thanks for the details - then agreed - I'll just update it now here, save opening another issue